lib/pagseguro/collection_managers.rb

```diff
@@ -0,0 +1,29 @@
+module PagSeguro
+  module CollectionManagers
+    module MacroMethods
+      def collection_type(klass)
+        define_method 'collection_type' do
+          self.class.classify(klass, PagSeguro)
+        end
+      end
+
+      def collection_manager(klass)
+        define_method 'collection_manager' do
+          self.class.classify(klass, CollectionManagers)
+        end
+      end
+
+      def camelize(param)
+        param.to_s.split('_').map(&:capitalize).join
+      end
+
+      def to_class(klass, super_klass)
+        super_klass.const_get( camelize(klass) )
```

Comment:
yes, sometime I put the blank spaces to improve the legibility, but I can take off...